src/java.base/share/classes/java/security/Security.java

```diff
@@ -144,6 +144,10 @@ private static void initialize() {
         // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
         if (InternalCRIUSupport.isCheckpointAllowed()) {
             CRIUConfigurator.setCRIUSecMode(props);
+            // Check if `-XX:+EnableFullSecurityEnv` is enabled, if it is then reconfigure the security providers.
+            if (InternalCRIUSupport.disableCRIUSecProvider()) {
+                CRIUConfigurator.setCRIURestoreMode();
+            }
         }
```

Comment:
I'm not convinced that this sequence of installing `CRIUSecProvider` and then removing it yields the same behavior as not installing it in the first place. I expect we want something like this:
```java
        if (InternalCRIUSupport.isCheckpointAllowed()
            && !InternalCRIUSupport.disableCRIUSecProvider()
        ) {
            CRIUConfigurator.setCRIUSecMode(props);
        }
```
Comments should be update to use the actual option name.

Reply:
updated